dashboard/client/src/pages/serve/hook/useServeApplications.ts

```diff
@@ -206,10 +212,32 @@ export const useServeHTTPProxyDetails = (httpProxyId: string | undefined) => {
   const httpProxy = httpProxyId ? data?.http_proxies?.[httpProxyId] : undefined;
 
   // Need to expose loading because it's not clear if undefined values
-  // for application, deployment, or replica means loading or missing data.
+  // for http proxies means loading or missing data.
   return {
     loading: isLoading,
     httpProxy,
     error,
   };
 };
+
+export const useServeControllerDetails = () => {
+  const { data, error, isLoading } = useSWR(
+    "useServeControllerDetails",
+    async () => {
+      const rsp = await getServeApplications();
+
+      if (rsp) {
+        return rsp.data;
+      }
+    },
+    { refreshInterval: API_REFRESH_INTERVAL_MS },
+  );
```

Comment:
nit: this should be extracted to a general `use(Get/Fetch)ServeApplications` hook, i see it repeated a fair bit